feat: improved text tracks settings labels

wseymour15 opened the pull request:
## Description
Based off of an audit that a Brightcove client received, we are updating the text tracks settings menu with improved labels. This change is regarding Web Content Accessibility Guidelines (WCAG). 

![Screen Shot 2023-01-31 at 3 08 32 PM](https://user-images.githubusercontent.com/26021721/215891069-cb9d1a90-78c8-400a-afcd-2987016144f5.png)


## Specific Changes proposed
- Labels added/improved to fields in the settings menu
- CSS so labels are displayed next to inputs
- Translations for labels
- Updates docs regarding missing translations.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chrome, Firefox, IE)
  - [ ] Unit Tests updated or fixed
  - [x] Docs/guides updated
  - [ ] Example created ([starter template on JSBin](https://codepen.io/gkatsev/pen/GwZegv?editors=1000#0))
- [ ] Reviewed by Two Core Contributors

gkatsev commented:
Not that it shouldn't change, but the names are taken from the CVAA, which has these user settings requirements.

OwenEdwards commented:
Personally, I think the text changes are fine. But my concern with the layout is that the "Font Size", "Text Edge Style", and "Font Family" controls are so close to the three "Opacity" controls that it isn't clear visually that the "Opacity" fields are each part of a fieldset with the legend "Text", "Text Background", and "Caption Area Background". That could be fixed by increasing the spacing, or adding an outline for each fieldset.

wseymour15 commented:
> That could be fixed by increasing the spacing, or adding an outline for each fieldset.

Good suggestion, I updated the CSS a bit to increase the spacing between some of these fields:

<img width="638" alt="Screen Shot 2023-02-01 at 10 17 25 AM" src="https://user-images.githubusercontent.com/26021721/216100634-2223cc5c-9525-4d26-ae41-c6a5619408c9.png">

mister-ben commented:
It looks quite cramped at a smallish size. It might look better with color and opacity to be slightly indented at this size. Maybe text/text background in as bold heading?

<img width="639" alt="grafik" src="https://user-images.githubusercontent.com/1676039/216166896-6a19cd5f-dff9-42ab-8e28-409da2bf5a73.png">

mister-ben commented on lang/de.json:
```diff
@@ -86,6 +86,10 @@
-  "No content": "Kein Inhalt"
+  "No content": "Kein Inhalt",
+  "Color": "Farbe",
+  "Opacity": "Opazität",
+  "Text Background": "Text-Hintergrund",
```
```suggestion
  "Opacity": "Deckkraft",
  "Text Background": "Texthintergrund",
```

mister-ben commented on lang/en.json:
```diff
@@ -87,5 +87,9 @@
   "Picture-in-Picture": "Picture-in-Picture",
-  "No content": "No content"
+  "No content": "No content",
+  "Color": "Color",
```
We're going to need an en-GB :)

wseymour15 replied:
Good call, created this issue here: https://github.com/videojs/video.js/issues/8105

gkatsev replied:
I think mastodon using british naming balances things out 😉

wseymour15 commented:
> It looks quite cramped at a smallish size. It might look better with color and opacity to be slightly indented at this size. Maybe text/text background in as bold heading?
You're right, that does not look as expected, and nor did a lot of mobile devices when the menu was opened full screen. I updated this with bolded headers and a few small styling changes to make it look better in most cases. The only flaw I found is that in some medium sized players, the user is now forced to scroll to get to the bottom of the menu. If this is an issue let me know, but I believe in most cases this will look much better.
<img width="639" alt="Screen Shot 2023-02-02 at 1 14 20 PM" src="https://user-images.githubusercontent.com/26021721/216430276-3e38abf1-40c4-4594-a00b-2953abb3c209.png">

gkatsev commented:
I think it's fine for now, even with the scrolling. I built it out 10 years and 2 months ago, and it needs to be re-designed from scratch at this point.

roman-bc-dev commented:
I agree with @mister-ben that a small indent to the left of the labeled settings (color, opacity) would also improve at-a-glance legibility. Overall, it looks much cleaner!

wseymour15 commented:
> I agree with @mister-ben that a small indent to the left of the labeled settings (color, opacity) would also improve at-a-glance legibility. Overall, it looks much cleaner!

@roman-bc-dev This was more work then I expected 😅 One of these label/select elements was wrapped in a span, the other was not.. so I updated the HTML to ensure that both are wrapped in a span element, updated the css, and also a few test fixes :)
